src/traces/image/attributes.js

```diff
@@ -0,0 +1,84 @@
+/**
+* Copyright 2012-2019, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var plotAttrs = require('../../plots/attributes');
+var hovertemplateAttrs = require('../../plots/template_attributes').hovertemplateAttrs;
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = extendFlat({
+    z: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'A 2-dimensional array in which each element is an array of 3 or 4 numbers representing a color.',
+        ].join(' ')
+    },
+    colormodel: {
+        valType: 'enumerated',
+        values: ['rgb', 'rgba', 'hsl', 'hsla'],
+        dflt: 'rgb',
+        role: 'info',
+        editType: 'plot',
+        description: 'Color model used to map the numerical color components described in `z` into colors.'
+    },
+    zmin: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'plot',
+        description: [
+            'Array defining the lower bound for each color component.',
+            'For example, for the `rgba` colormodel, the default value is [0, 0, 0, 0].'
+        ].join(' ')
+    },
+    zmax: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'plot',
+        description: [
+            'Array defining the higher bound for each color component.',
+            'For example, for the `rgba` colormodel, the default value is [255, 255, 255, 1].'
+        ].join(' ')
+    },
+    x0: {
+        valType: 'number',
+        dflt: 0,
+        role: 'info',
+        editType: 'calc',
```

Comment:
You might need `editType: 'calc+clearAxisTypes'` for `x0` and `y0` in order to restyle to work properly in situations like:

```js
Plotly.newPlot(gd, [{
  z: /**/,
  x0: 2
}])

Plotly.restyle(gd, 'x0', 'a')
```

Reply:
... moreover `x0` and `y0` should be `valType: 'any'` to support categorical axes.

Reply:
... well I set `valType: 'any'`. I didn't change the `editType` however. Let me check if it's really necessary.

Reply:
It was needed indeed. Thanks @etpinard :1st_place_medal: 

Fixed in https://github.com/plotly/plotly.js/pull/4289/commits/0eb6604ab3470c28ee866d4cb2f90087ce81d7db

Reply:
Ok - can you add a test to lock it down? thanks.

Reply:
Of course :smile_cat: I added one in e2a9c6e77a425c96d2878168c204427d7460d797

Reply:
Thanks!!